src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -291,17 +291,58 @@ public static void LogServiceBusDependency(this ILogger logger, string entityNam
+
+            context = context ?? new Dictionary<string, object>();
 
             logger.LogInformation(ServiceBusDependencyFormat, "Azure Service Bus", entityType, entityName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
         }
 
+        /// <summary>
+        ///     Logs an Azure Table Storage Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="accountName">Account of the storage resource</param>
+        /// <param name="tableName">Name of the Table Storage resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="measurement">Measuring the latency to call the Table Storage dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogTableStorageDependency(this ILogger logger, string accountName, string tableName, bool isSuccessful, DependencyMeasurement measurement, Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            Guard.NotNullOrWhitespace(tableName, nameof(tableName));
+            Guard.NotNullOrWhitespace(accountName, nameof(accountName));
+            Guard.NotNull(measurement, nameof(measurement));
+
+            LogTableStorageDependency(logger, accountName: accountName, tableName: tableName, isSuccessful: isSuccessful, startTime: measurement.StartTime, duration: measurement.Elapsed, context: context);
+        }
+
+        /// <summary>
+        ///     Logs an Azure Table Storage Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="accountName">Account of the storage resource</param>
+        /// <param name="tableName">Name of the Table Storage resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="startTime">Point in time when the interaction with the dependency was started</param>
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogTableStorageDependency(this ILogger logger, string accountName, string tableName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            Guard.NotNullOrWhitespace(tableName, nameof(tableName));
+            Guard.NotNullOrWhitespace(accountName, nameof(accountName));
+
+            logger.LogInformation(DependencyFormat, "Azure Table Storage", tableName, accountName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
```

Comment:
I was wrong and should actually be this:
```suggestion
            logger.LogInformation(DependencyFormat, "Azure table", tableName, accountName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
```

See https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/WEB/Src/DependencyCollector/Shared/Implementation/RemoteDependencyConstants.cs

One option is to add a reference for sake of using these constants but they are internal. Alternative is copying it over but we could then also just stick with our approach.

Reply:
Changed and I think we should:
1. Not point to the SDK and stick with our approach
2. But move our dependency types to DependencyTypes.cs which contains all the constants maybe to consolidate it and improve discoverability?

Thoughts @stijnmoreels?